kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveKafkaStreams.java

```diff
@@ -131,24 +134,41 @@ private static StreamsConfig verifiedStreamsConfigs(
 
     // TODO(sophie): finish writing KIP to make this a public StreamsConfig, it's a bit awkward to
     //  be asking users to set an internal config and very very rare to be set to anything else
-    final String taskAssignor = (String) configs.get(InternalConfig.INTERNAL_TASK_ASSIGNOR_CLASS);
-    if (!TASK_ASSIGNOR_CLASS_OVERRIDE.equals(taskAssignor)) {
+    final Object o = configs.get(InternalConfig.INTERNAL_TASK_ASSIGNOR_CLASS);
+    if (o == null) {
       final String errorMsg = String.format(
-          "Invalid Streams configuration '%s': please override to '%s'",
+          "Invalid Streams configuration value for '%s': please override to '%s'",
           InternalConfig.INTERNAL_TASK_ASSIGNOR_CLASS,
           TASK_ASSIGNOR_CLASS_OVERRIDE
       );
       LOG.error(errorMsg);
       throw new ConfigException(errorMsg);
-    } else {
-      propsWithOverrides.put(
-          InternalConfig.INTERNAL_TASK_ASSIGNOR_CLASS, TASK_ASSIGNOR_CLASS_OVERRIDE
+    }
+
+    if (!TASK_ASSIGNOR_CLASS_OVERRIDE.equals(o.toString())) {
```

Comment:
They're separate because I wanted to give different error messages. But in light of this, doesn't matter:

> I'm OK overriding this if they passed in null

Ah must have misremembered the outcome of that discussion 😅 Glad to hear though